src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx

```diff
@@ -190,68 +191,96 @@ function calculateOverallProgress(index: IndexSharedInfo) {
-            </ProgressCircle>
-        );
-    }
-
-    if (index.nodesInfo.some((x) => x.details?.status === "Disabled")) {
-        return (
-            <ProgressCircle inline state="running" icon={iconForState("Disabled")}>
-                Disabled
-            </ProgressCircle>
-        );
-    }
-
-    if (index.nodesInfo.some((x) => x.details?.status === "Paused")) {
-        return (
-            <ProgressCircle inline state="running" icon={iconForState("Paused")}>
-                Paused
-            </ProgressCircle>
-        );
-    }
-
-    if (index.nodesInfo.some((x) => x.details?.status === "Pending")) {
-        return (
-            <ProgressCircle inline state="running" icon={iconForState("Pending")}>
-                Pending
-            </ProgressCircle>
-        );
-    }
-
-    if (index.nodesInfo.some((x) => x.details?.stale)) {
-        const overallProgress = calculateOverallProgress(index);
-
-        return (
-            <ProgressCircle inline state="running" progress={overallProgress}>
-                Running
-            </ProgressCircle>
-        );
-    }
+    const { index, onClick } = props;
+    const getState = (index: IndexSharedInfo) => {
+        if (index.nodesInfo.some((x) => x.status === "failure" || x.details?.faulty || x.details?.state === "Error")) {
+            return "failed";
+        }
+        if (
+            index.nodesInfo.some(
+                (x) =>
+                    x.details?.status === "Disabled" ||
+                    x.details?.status === "Paused" ||
+                    x.details?.status === "Pending" ||
+                    x.details?.stale
+            )
+        ) {
+            return "running";
+        }
+        return "success";
+    };
+
+    const getIcon = (index: IndexSharedInfo): IconName => {
+        if (index.nodesInfo.some((x) => x.status === "failure" || x.details?.faulty || x.details.state === "Error")) {
+            return "cancel";
+        }
+        if (index.nodesInfo.some((x) => x.details?.status === "Disabled")) {
+            return iconForState("Disabled");
+        }
+        if (index.nodesInfo.some((x) => x.details?.status === "Paused")) {
+            return iconForState("Paused");
+        }
+        if (index.nodesInfo.some((x) => x.details?.status === "Pending")) {
+            return iconForState("Pending");
+        }
+        if (index.nodesInfo.some((x) => x.details?.stale)) {
+            return null;
+        }
+
+        return "check";
+    };
+
+    const getStateText = (index: IndexSharedInfo) => {
+        if (index.nodesInfo.some((x) => x.status === "failure")) {
+            return "Load errors";
+        }
+        if (index.nodesInfo.every((x) => x.details?.faulty)) {
+            return "Faulty";
+        }
+        if (index.nodesInfo.some((x) => x.details?.faulty)) {
+            return "Some faulty";
+        }
+        if (index.nodesInfo.every((x) => x.details?.state === "Error")) {
+            return "Errors";
```

Comment:
it's plural?

Reply:
It makes sense for multiple errors:
![image](https://user-images.githubusercontent.com/21218477/236481330-05006db5-4899-4f42-97fd-09474c5c8650.png)